src/sentry/eventstore/base.py

```diff
@@ -192,3 +194,12 @@ def bind_nodes(self, object_list, node_name="data"):
         for item, node in object_node_list:
             data = node_results.get(node.id) or {}
             node.bind_data(data, ref=node.get_ref(item))
+
+    def generate_node_id(self, project_id, event_id):
```

Comment:
Why not putting it in NodeData or in nodestore ? This seems conceptually part of nodestore, more than eventstore.

Reply:
I kind of struggled with where to put this.

My thinking was:

- I don't think it's part of nodestore or NodeData because 1. nodestore doesn't know anything about event IDs or project IDs (just gets / sets data based on a provided ID that is generated elsewhere). 2. RawEvent uses NodeData in the same way as our Event object and there it is always being passed a generated ID.

- Keeping it where it is now on Event (or more precisely EventCommon right now) is probably the most consistent with RawEvent. I moved it here because I thought it'll be slightly nicer to avoid importing the function from its new home in sentry.eventstore.models.Event. I think it's also better to slim down the event object where possible since we pass it around a lot and need to pickle/unpickle, etc.

- Having a method on eventstore seemed like the next logical choice